apps/minifront/src/components/send/send-form/index.tsx

```diff
@@ -31,11 +35,42 @@ export const SendForm = () => {
     setFeeTier,
     setMemo,
     sendTx,
+    setGasPrices,
+    setStakingToken,
     txInProgress,
   } = useStore(sendSelector);
 
   useRefreshFee();
 
+  // Determine if the selected token is the staking token based on the current balances and metadata
+  const isStakingToken = hasStakingToken(
+    transferableBalancesResponses?.data,
+    stakingTokenMetadata.data,
+    selection,
+  );
+
+  // useEffect here defers the state updates until after the rendering phase is complete,
+  // preventing direct state modifications during rendering.
+  useEffect(() => {
+    const updateStakingTokenAndGasPrices = () => {
+      // Update the zustand store and local state
+      setStakingToken(isStakingToken);
+      if (gasPrices.data) {
+        setGasPrices(gasPrices.data);
+      }
+    };
+
+    updateStakingTokenAndGasPrices();
+  }, [
+    transferableBalancesResponses,
+    stakingTokenMetadata,
+    selection,
+    gasPrices,
+    isStakingToken,
+    setGasPrices,
+    setStakingToken,
+  ]);
```

Comment:
issue: It appears this logic is not critical for the display of this component. This looks to be only used in `assembleRequest()` right? Think you should move it entirely out of react-land and put it there so you don't need to worry about the hooks stuff.

Reply:
modified the useEffect in the react component to 

```
  useEffect(() => {
    const updateStakingToken = () => {
      // Update the zustand store and local state
      setStakingToken(isStakingToken);
    };

    updateStakingToken();
  }, [
    transferableBalancesResponses,
    stakingTokenMetadata,
    selection,
    isStakingToken,
    setStakingToken,
  ]);
```

since `isStakingToken` is relevant for the `NonNativeFeeWarning`, it makes sense to set the state if we’re already computing it. Moving this logic into `assembleRequest` would require access to `hasStakingToken()`, which in turn would necessitate creating a dedicated Zustand slice. As noted below, that approach wasn’t working effectively.